src/core/energy_inline.hpp

```diff
@@ -68,12 +67,16 @@
 #include "electrostatics_magnetostatics/mmm2d.hpp"
```

Comment:
I think those explicit includes are not needed any more, `coulomb.hpp` and `dipolar.hpp` should be sufficient.